web-server/src/components/Service/SystemLog/LogSearch.tsx

```diff
@@ -0,0 +1,179 @@
+  Button,
+  InputAdornment,
+  TextField,
+  Typography,
+  Box
+} from '@mui/material';
+import { styled } from '@mui/material/styles';
+import debounce from 'lodash/debounce';
+import { memo, useState, useCallback, useMemo, useEffect } from 'react';
+
+import { MotionBox } from '@/components/MotionComponents';
+
+const SearchContainer = styled('div')(() => ({
+  position: 'sticky',
+  top: 0,
+  zIndex: 1,
+  gap: 5,
+  paddingBottom: 8,
+  alignItems: 'center',
+  backdropFilter: 'blur(10px)',
+  borderRadius: 5
+}));
+
+const SearchControls = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  gap: theme.spacing(1),
+  marginTop: 8
+}));
+
+const StyledTextField = styled(TextField)(({ theme }) => ({
+  '& .MuiOutlinedInput-root': {
+    backgroundColor: theme.palette.background.paper,
+    transition: 'all 0.2s ease-in-out',
+    '&:hover': {
+      backgroundColor: theme.palette.background.paper,
+      boxShadow: `0 0 0 1px ${theme.palette.primary.main}`
+    },
+    '&.Mui-focused': {
+      backgroundColor: theme.palette.background.paper,
+      boxShadow: `0 0 0 2px ${theme.palette.primary.main}`
+    }
+  }
+}));
+
+interface LogSearchProps {
+  onSearch: (query: string) => void;
+  onNavigate: (direction: 'prev' | 'next') => void;
+  currentMatch: number;
+  totalMatches: number;
+}
+
+const LogSearch = memo(({
+  onSearch,
+  onNavigate,
+  currentMatch,
+  totalMatches
+}: LogSearchProps) => {
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const debouncedSearch = useMemo(
+    () =>
+      debounce((query: string) => {
+        onSearch(query);
+      }, 300),
+    [onSearch]
+  );
+
+  useEffect(() => {
+    return () => {
+      debouncedSearch.cancel();
+    };
+  }, [debouncedSearch]);
+
+  const handleSearchChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const query = event.target.value;
+      setSearchQuery(query);
+      debouncedSearch(query);
+    },
+    [debouncedSearch]
+  );
+
+  const handleClear = useCallback(() => {
+    setSearchQuery('');
+    onSearch('');
+    debouncedSearch.cancel();
+  }, [onSearch, debouncedSearch]);
+
+  const handleNavigate = useCallback(
+    (direction: 'prev' | 'next') => {
+      onNavigate(direction);
+    },
+    [onNavigate]
+  );
+
+  const showSearchControls = useMemo(
+    () => searchQuery && totalMatches > 0,
+    [searchQuery, totalMatches]
+  );
+
+  return (
+    <SearchContainer>
+      <StyledTextField
+        fullWidth
+        variant="outlined"
+        placeholder="Search logs..."
+        value={searchQuery}
+        onChange={handleSearchChange}
+        InputProps={{
+          startAdornment: (
+            <InputAdornment position="start">
+              <SearchIcon color="action" />
+            </InputAdornment>
+          ),
+          endAdornment: searchQuery && (
+            <InputAdornment position="end">
+              <ClearIcon
+                style={{ cursor: 'pointer' }}
```

Comment:
Prefer using MUI's `sx` prop.
It allows using MUI's css-in-js optimizations and the final result doesn't include inline-styles as a result.

Common styles are combined into the same generated classname to keep the bundle-size small.

Reply:
Do this for any other instances where it's an MUI component.